### PR TITLE
Add ability to upload a file to liquibase script generator

### LINKED ISSUE #1484: Allow to upload/edit databasechangeset in Liquibase script generator
**Reporter**
Gerrit

**Describe the solution you'd like**
It would be nice to have the ability to view/edit/upload a liquibase databasechangeset to the Liquibase script generator. This would help application updates, as it would allow to do/prepare the database changes independent of the actual deployment.
It would also allow for a workaround for the problem that Liquibase checksums can change over Liquibase versions.

### PR DESCRIPTION
**Pull request by alisihab**
Closes #1484

### CODE REVIEW

File: webapp/src/main/webapp/iaf/gui/js/controllers.js [hunk 1]
```diff
@@ -1782,7 +1749,7 @@ angular.module('iaf.beheerconsole')
 
 .controller('AddScheduleCtrl', ['$scope', 'Api', 'Misc', function($scope, Api, Misc) {
 	$scope.state = [];
-	$scope.addLocalAlert = function(type, message) {
```
Comment (nielsm5) [inline webapp/src/main/webapp/iaf/gui/js/controllers.js:1785]: I think a merge went wrong here..

**Review by nielsm5 — CHANGES_REQUESTED** [review]

File: core/src/main/java/nl/nn/adapterframework/jdbc/migration/LiquibaseLogger.java [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+	public void setLogLevel(String logLevel, String logFile) {
+	}
+
+	@Override
+	public void setChangeLog(DatabaseChangeLog databaseChangeLog) {
+	}
+
+	@Override
+	public void setChangeSet(ChangeSet changeSet) {
+	}
```
Comment (nielsm5) [inline core/src/main/java/nl/nn/adapterframework/jdbc/migration/LiquibaseLogger.java:98]: Are you purposely overriding these? If so, please give a brief explanation as to why. Else please remove these methods, as I can imagine it's handy to know which file/changeset contains an error.

File: core/src/main/java/nl/nn/adapterframework/jdbc/migration/LiquibaseLogger.java [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+	@Override
+	public void debug(String message, Throwable e) {
+		if(log.isDebugEnabled()) {
+			log.debug(message, e);
+		}
+	}
+
+	@Override
+	public void setLogLevel(String logLevel, String logFile) {
+	}
```
Comment (nielsm5) [inline core/src/main/java/nl/nn/adapterframework/jdbc/migration/LiquibaseLogger.java:90]: I can imagine when you delegate everything to log4j you either want to let log4j handle this (and set the loglevel default to INFO, as per the old log4j4ibis.xml).

File: core/src/main/java/nl/nn/adapterframework/jdbc/migration/LiquibaseLogger.java [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+		if(log.isTraceEnabled()) {// prints too much unnecessary only print if trace enabled
+			log.debug(message);
+		}
+	}
+
+	@Override
+	public void debug(String message, Throwable e) {
+		if(log.isDebugEnabled()) {
+			log.debug(message, e);
+		}
```
Comment (nielsm5) [inline core/src/main/java/nl/nn/adapterframework/jdbc/migration/LiquibaseLogger.java:85]: You don't need these if statements, it's the first thing log4j does :)

File: core/src/main/java/nl/nn/adapterframework/jdbc/migration/LiquibaseLogger.java [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+/**
+ * @author alisihab
+ *
+ */
+public class LiquibaseLogger extends AbstractLogger {
+	private static final Logger log = LogUtil.getLogger(LiquibaseLogger.class);
+
+	@Override
+	public void setName(String name) {
+	}
```
Comment (nielsm5) [inline core/src/main/java/nl/nn/adapterframework/jdbc/migration/LiquibaseLogger.java:34]: Give a brief explanation that calling this wont do much as the loggers name has already been defined upon creating the logger

File: core/src/main/java/nl/nn/adapterframework/jdbc/migration/LiquibaseLogger.java [hunk 6]
```diff
@@ -0,0 +1,103 @@
...
+
+	@Override
+	public void severe(String message) {
+	}
```
Comment (nielsm5) [inline core/src/main/java/nl/nn/adapterframework/jdbc/migration/LiquibaseLogger.java]: I think these should also be logged?

File: core/src/main/java/nl/nn/adapterframework/jdbc/migration/LiquibaseImpl.java [hunk 7]
```diff
@@ -71,6 +72,16 @@ public LiquibaseImpl(DataSource datasource, Configuration configuration, String
 		}
 		connection = new JdbcConnection(datasource.getConnection());
 
+		// Override liquibase internal logger
+		LogFactory.setInstance(new LogFactory() {
```
Comment (nielsm5) [inline core/src/main/java/nl/nn/adapterframework/jdbc/migration/LiquibaseImpl.java:76]: Please override getLog(String) instead to be able to fetch the correct logger.

File: core/src/main/java/nl/nn/adapterframework/jdbc/migration/LiquibaseLogger.java [hunk 8]
```diff
@@ -0,0 +1,103 @@
...
+import nl.nn.adapterframework.util.LogUtil;
+
+import org.apache.logging.log4j.Logger;
+
+/**
+ * @author alisihab
+ *
+ */
+public class LiquibaseLogger extends AbstractLogger {
+	private static final Logger log = LogUtil.getLogger(LiquibaseLogger.class);
```
Comment (nielsm5) [inline core/src/main/java/nl/nn/adapterframework/jdbc/migration/LiquibaseLogger.java:30]: Please give each logger a name.
Right now all log statements come from the migration.LiquibaseLogger package, which hides it's origin.
You can set the final field in a constructor where you may pass the name of the logger.

**Review by nielsm5 — CHANGES_REQUESTED** [review]

**Review by nielsm5 — APPROVED** [review]

**Review by gvanbrakel — APPROVED** [review]

**Pull request merged**